app/components/developer_query_component.html.erb

```diff
@@ -83,8 +83,88 @@
+        </button>
+      </div>
+      <div data-toggle-target="element" class="bg-gray-50">
+        <div class="fixed inset-0 flex z-40 sm:hidden" role="dialog" aria-modal="true">
+          <div class="fixed inset-0 bg-black bg-opacity-25" aria-hidden="true"></div>
+          <div class="ml-auto relative max-w-xs w-full h-full bg-white shadow-xl py-4 pb-6 flex flex-col overflow-y-auto">
+            <div class="px-4 mb-2 flex items-center justify-between">
+              <h2 class="text-lg font-medium text-gray-900">Filters</h2>
+              <button type="button" data-action="toggle#toggle" class="-mr-2 w-10 h-10 bg-white p-2 rounded-md flex items-center justify-center text-gray-400 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-indigo-500">
+                <span class="sr-only">Close menu</span>
+                <%= inline_svg_tag "icons/outline/x.svg", class: "h-6 w-6" %>
+              </button>
+            </div>
+            <%= form_with url: developers_path, method: :get, html: {class: "py-6"} do |form| %>
+
+              <div class="border-t border-gray-200 px-4 py-6" data-controller="toggle" data-toggle-close-class="hidden">
+                <h3 class="-mx-2 -my-3 flow-root">
+                  <button type="button" data-action="toggle#toggle" class="px-2 py-3 bg-white w-full flex items-center justify-between text-sm text-gray-400" aria-controls="filter-section-0" aria-expanded="false">
+                    <span class="font-medium text-gray-900"><%= t(".work_preference.title") %></span>
+                    <span class="ml-6 flex items-center">
+                      <%= inline_svg_tag "icons/solid/chevron_down.svg", class: "rotate-0 h-5 w-5 transform", data: {"toggle-target": "chevron"} %>
+                    </span>
+                  </button>
+                </h3>
+
+                <div class="pt-6" id="filter-section-0" data-toggle-target="element">
+                  <div class="space-y-6">
+                    <%= form.collection_check_boxes(:role_types, role_types, :first, :last, {include_hidden: false}) do |b| %>
+                      <div class="flex items-center">
+                        <%= b.check_box class: "h-4 w-4 border-gray-300 rounded text-gray-600 focus:ring-gray-500", id: "mobile_role_types_#{b.object.first}", checked: role_selected?(b.object) %>
+                        <%= b.label class: "ml-3 pr-6 text-sm font-medium text-gray-900 whitespace-nowrap", for: "mobile_role_types_#{b.object.first}" %>
+                      </div>
+                    <% end %>
+                    <hr>
+                    <div class="flex items-top">
+                      <%= form.check_box :include_not_interested, class: "mt-1 h-4 w-4 border-gray-300 rounded text-gray-600 focus:ring-gray-500", include_hidden: false, checked: include_not_interested? %>
+                      <%= form.label :include_not_interested, t(".include_not_interested"), class: "ml-3 pr-6 text-sm font-medium text-gray-900" %>
```

Comment:
This needs the same `mobile_...` manually setting of the ID so we can tap the label to toggle the checkbox.